src/providers/mbed_provider/generate_random.rs

```diff
@@ -0,0 +1,27 @@
+use super::MbedProvider;
+use crate::authenticators::ApplicationName;
+use parsec_interface::operations::psa_generate_random;
+use parsec_interface::requests::{ResponseStatus, Result};
+use psa_crypto::operations::other::generate_random;
+
+impl MbedProvider {
+    pub(super) fn psa_generate_random_internal(
+        &self,
+        _app_name: ApplicationName,
+        op: psa_generate_random::Operation,
+    ) -> Result<psa_generate_random::Result> {
+        let buffer_size = op.size;
+        let mut buffer = vec![0u8; buffer_size];
+
+        match generate_random(&mut buffer) {
+            Ok(_) => Ok(psa_generate_random::Result {
+                random_bytes: buffer.into(),
+            }),
+            Err(error) => {
+                let error = ResponseStatus::from(error);
+                format_error!("Generate random status: ", error);
```

Comment:
```suggestion
                format_error!("Generate random status", error);
```

The `format_error!` macro adds the `: ` for you, but maybe I should change that since everyone seems tempted to use it as the normal `error!`